### PR TITLE
[mesheryctl] Refactor kubeconfig setting logic in `config.go`

### LINKED ISSUE #4257: [mesheryctl] Error during installation on Ubuntu
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
I got an error installing mesheryctl / meshery on an ubuntu VM

#### Screenshots/Logs
````
 curl -L https://git.io/meshery | PLATFORM=kubernetes bash -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  6674  100  6674    0     0  13933      0 --:--:-- --:--:-- --:--:-- 13933

Downloading mesheryctl-v0.5.60 for Linux...

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   642  100   642    0     0   2301      0 --:--:-- --:--:-- --:--:--  2301
100 12.4M  100 12.4M    0     0  14.7M      0 --:--:-- --:--:-- --:--:-- 96.3M

Extracting mesheryctl-v0.5.60 to temporary folder y5WRdDJEvRwxAs3m...
Archive:  /home/ubuntu/meshery.zip
  inflating: /home/ubuntu/y5WRdDJEvRwxAs3m/LICENSE
  inflating: /home/ubuntu/y5WRdDJEvRwxAs3m/README.md
  inflating: /home/ubuntu/y5WRdDJEvRwxAs3m/mesheryctl

Installing mesheryctl in /usr/local/bin.
Permission Resolved: mesheryctl installed using sudo permissions.
permissions moved to ubuntu
Removing installation files
Starting Meshery...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1959294]

goroutine 1 [running]:
github.com/layer5io/meshery/mesheryctl/internal/cli/root.initConfig()
	/Users/runner/work/meshery/meshery/mesheryctl/internal/cli/root/root.go:112 +0x894
github.com/spf13/cobra.(*Command).preRun(0x2d044c0)
	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:886 +0x49
github.com/spf13/cobra.(*Command).execute(0x2d044c0, 0xc0004481b0, 0x3, 0x3, 0x2d044c0, 0xc0004481b0)
	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:822 +0x14f
github.com/spf13/cobra.(*Command).ExecuteC(0x2cfdbc0, 0x201e680, 0x2ee5488, 0x0)
	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
github.com/layer5io/meshery/mesheryctl/internal/cli/root.Execute()
	/Users/runner/work/meshery/meshery/mesheryctl/internal/cli/root/root.go:68 +0x56
main.main()
	/Users/runner/work/meshery/meshery/mesheryctl/cmd/mesheryctl/main.go:22 +0x25
````

#### Environment

- Host OS: Running an HyperKit Ubuntu VM on MacOS
- Platform: Kubernetes / K3s

---

#### Contributor Resources
- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

**Description**

This PR fixes #4257 

Moves setting the path to kubeconfig file to the `root.go` file and would be initialised before running every command.

Removed the logic for creating `.meshery` folder out of `config.go` and would be centrally done in `root.go`.

Sticking to this behaviour: If the `.meshery` folder or the `config.yaml` file is missing, it will prompt the user to create a new config file. This would be handled centrally in `root.go` and would not be repeated in `config.go`.

Modifies the changes made here: https://github.com/meshery/meshery/pull/4183

**Notes for Reviewers**

If this PR is tested and merged, please make a new release ASAP.

- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/pkg/utils/helpers.go [hunk 1]
```diff
@@ -102,6 +102,10 @@ var (
 	// ServiceAccount is the name of a Kubernetes manifest file required to setup Meshery
 	// check https://github.com/layer5io/meshery/tree/master/install/deployment_yamls/k8s
 	ServiceAccount = "service-account.yaml"
+	// To upload with param name
+	ParamName = "k8sfile"
+	// kubeconfig file name
+	KubeConfigYaml = "kubeconfig.yaml"
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/helpers.go:108]: Is this a temp file?

**Comment by leecalcote** [comment]
Were you able to check the other subcommands for their use of cobra inits() in their inits()?

**Comment by hexxdump** [comment]
@navendu-pottekkat, Its not clear on which function takes care of creating .meshery folder? root.go or any other util function?

File: mesheryctl/pkg/utils/platform.go [hunk 2]
```diff
@@ -768,3 +769,16 @@ func InstallprereqDocker() error {
 	log.Info("Prerequisite Docker Compose is installed.")
 	return nil
 }
+
+// Sets the path to user's kubeconfig file into global variables
+func SetKubeConfig() {
```
Comment (hexxdump) [inline mesheryctl/pkg/utils/platform.go:774]: Moving the logic to common utils code makes sense. Assuming .meshery folder is created already at this point of time.

Comment (navendu-pottekkat) [reply mesheryctl/pkg/utils/platform.go:774]: Yes. mesheryctl will create the `.meshery` folder and a default meshconfig file if it is missing while running **any** command.

**Comment by navendu-pottekkat** [comment]
@hexxdump The creation of the `.meshery` folder if it is not present and adding a template config is handled in `root.go`.

**Pull request merged**